src/main/java/org/alfresco/repo/bulkimport/metadataloaders/AbstractMapBasedMetadataLoader.java

```diff
@@ -101,68 +101,85 @@ public final String getMetadataFileExtension()
-                        for (final String aspectName : aspectNames)
+                        if (propertyDefinition.isMultiValued())
                         {
-                            QName aspect = QName.createQName(aspectName.trim(), namespaceService);
-                            metadata.addAspect(aspect);
+                            // Multi-valued property
+                            ArrayList<Serializable> values = new ArrayList<Serializable>(
+                                    Arrays.asList(((String) metadataProperties.get(key)).split(multiValuedSeparator)));
+                            metadata.addProperty(name, values);
+                        }
+                        else
+                        {
+                            // Single value property
+                            metadata.addProperty(name, metadataProperties.get(key));
                         }
                     }
-                    else  // Any other key => property
+                    else
                     {
-                        //####TODO: figure out how to handle properties of type cm:content - they need to be streamed in via a Writer 
-                    	QName              name               = QName.createQName(key, namespaceService);
-                    	PropertyDefinition propertyDefinition = dictionaryService.getProperty(name);  // TODO: measure performance impact of this API call!!
-                    	
-                    	if (propertyDefinition != null)
-                    	{
-                        	if (propertyDefinition.isMultiValued())
-                        	{
-                                // Multi-valued property
-                        		ArrayList<Serializable> values = new ArrayList<Serializable>(Arrays.asList(((String)metadataProperties.get(key)).split(multiValuedSeparator)));
-                        	    metadata.addProperty(name, values);
-                        	}
-                        	else
-                        	{
-                        	    // Single value property
-                        		metadata.addProperty(name, metadataProperties.get(key));
-                        	}
-                    	}
-                    	else
-                    	{
-                    	    if (log.isWarnEnabled()) log.warn("Property " + String.valueOf(name) + " doesn't exist in the Data Dictionary.  Ignoring it.");
-                    	}
+                        if (log.isWarnEnabled())
+                            log.warn("Property " + String.valueOf(name) + " from '" + metadataFilePath
+                                    + "' doesn't exist in the Data Dictionary.  Ignoring it.");
                     }
                 }
             }
-            else
-            {
-                if (log.isWarnEnabled()) log.warn("Metadata file '" + FileUtils.getFileName(metadataFile) + "' is not readable.");
-            }
+        }
+        else
+        {
+            if (log.isWarnEnabled())

```

Comment:
We still need to use braces, please.
Please fix on {{develop}}.

Reply:
https://github.com/Alfresco/alfresco-repository/commit/1d56cf1b7cf715e70cbe1f580af164697f572901 done